bin/node-template/runtime/src/lib.rs

```diff
@@ -1015,6 +1025,162 @@ impl darwinia_tron_backing::Trait for Runtime {
 
 impl darwinia_header_mmr::Trait for Runtime {}
 
+// dvm needed
+/// Fixed gas price of `1`.
+pub struct FixedGasPrice;
+
+impl FeeCalculator for FixedGasPrice {
+	fn min_gas_price() -> U256 {
+		// Gas price is always one token per gas.
+		0.into()
+	}
+}
+
+parameter_types! {
+	pub const ChainId: u64 = 43;
+}
+
+pub struct ConcatAddressMapping;
+
+impl AddressMapping<AccountId> for ConcatAddressMapping {
+	fn into_account_id(address: H160) -> AccountId {
+		let suffix = &BlakeTwo256::hash(&address[..])[0..12];
+
+		let mut data = [0u8; 32];
+		data[0..20].copy_from_slice(&address[..]);
+		data[20..32].copy_from_slice(suffix);
+		AccountId::from(data)
+	}
+}
+
+fn ensure_linear_cost(
+	target_gas: Option<usize>,
+	len: usize,
+	base: usize,
+	word: usize,
+) -> Result<usize, ExitError> {
+	let cost = base
+		.checked_add(
+			word.checked_mul(len.saturating_add(31) / 32)
+				.ok_or(ExitError::OutOfGas)?,
+		)
+		.ok_or(ExitError::OutOfGas)?;
+
+	if let Some(target_gas) = target_gas {
+		if cost > target_gas {
+			return Err(ExitError::OutOfGas);
+		}
+	}
+
+	Ok(cost)
+}
+
+/// Precompile for withdrawing from evm address
+pub struct Withdraw;
```

Comment:
this withdraw precompiled contract might be required by other runtimes too, better to move to library part.

Reply:
I recommend moving this code piece to ethereum pallet under dvm.

Reply:
Done.